PP-4280 Reorganised Gateway (payment gateway) related classes

kbottla opened the pull request:
## WHAT
First in the line of PRs to restructure classes (through out the connector) into clearly identifiable top level functions/domains.

This change includes
- Organised payment gateway (worldpay, epdq, smartpay, sandbox) specific classes into `gateway` package and sub packages
- Mostly includes changes to package name & corresponding imports
- Qualifier changes for few static variables

Doesn't include
- Any changes to existing functionality

@mrlumbu

mrlumbu commented:
@alexbishop1  `gateway` is not a bad idea. I would like to standardise on one term (duh). Or maybe `paymentgateway`? Then use this everywhere - `PaymentProvider` makes no sense. We could just use `gateway` everywhere, but it feels a bit too generic, even the bounded context of connector

alexbishop1 commented:
I’m sure we could bikeshed on the names forever. The way connector alternates between “gateway” and “(payment) provider” is frustrating, though it seems to mainly use “gateway” (though often in the context of “gateway account”, which is a concept that we’ve allowed to expand to represent more than purely an account with a payment gateway.

I don’t think `gateway` is too generic for connector: it’s a fairly well-defined term within the domain of card payments.

The abbreviation `psp` is another option and has the appeal of being short.

mrlumbu commented:
Ok, let's go with `gateway`, if you're happy with that @kbottla?

kbottla commented:
That sounds good to me :)

mrlumbu commented:
In a future PR we'll use gateway everywhere that we currently use PaymentProvider and friends

kbottla commented:
Thanks @alexbishop1 
- I have renamed package to `gateway`
- Wild imports have been removed for organised packages
- Removed fully classified name in `UnknownStatusTest`
